zhmccli/_cmd_partition.py

```diff
@@ -916,6 +967,133 @@ def cmd_partition_console(cmd_ctx, cpc_name, partition_name, options):
+    }
+    options = original_options(options)
+    properties = options_to_properties(options, name_map)
+    volume_opt = options['volume']  # It is required
+    timeout = options['timeout']  # It is optional but has a default
+
+    if storage_management_feature(partition):
+
+        # Check and parse the --volume option into SG.SV or UUID
+        volume_parts = volume_opt.split('.')
+        if len(volume_parts) == 1:  # format: UUID
+            uuid = volume_parts[0]
+            sg_list = partition.list_attached_storage_groups()
+            storage_volume = None
+            for sg in sg_list:
+                try:
+                    storage_volume = sg.storage_volumes.find(uuid=uuid)
+                    break
+                except zhmcclient.NotFound:
+                    continue
+            if not storage_volume:
+                raise_click_exception(
+                    "Storage volume with UUID '{uuid}' specified in the "
+                    "--volume option was not found in any storage group "
+                    "attached to partition '{part}' of CPC '{cpc}'.".
+                    format(uuid=uuid, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
+        elif len(volume_parts) == 2:  # format: SG.SV
+            sg_name, sv_name = volume_parts
+            sg_list = partition.list_attached_storage_groups()
+            sg = None
+            for _sg in sg_list:
+                if _sg.name == sg_name:
+                    sg = _sg
+                    break
+            if not sg:
+                raise_click_exception(
+                    "Storage group '{sg}' specified in the --volume option was "
+                    "not found in the storage groups attached to partition "
+                    "'{part}' of CPC '{cpc}'.".
+                    format(sg=sg_name, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
+            try:
+                storage_volume = sg.storage_volumes.find(name=sv_name)
+            except zhmcclient.NotFound:
+                raise_click_exception(
+                    "Storage volume '{sv}' specified in the --volume option "
+                    "was not found in the volumes of storage group '{sg}' "
+                    "attached to partition '{part}' of CPC '{cpc}'.".
+                    format(sv=sv_name, sg=sg_name, part=partition_name,
+                           cpc=cpc_name),
+                    cmd_ctx.error_format)
+        else:
+            raise_click_exception(
+                "Invalid format for --volume option: '{v}'. CPC '{cpc}' has "
+                "the storage management feature which requires the formats "
+                "'SG.SV' or 'UUID'.".
+                format(v=volume_opt, cpc=cpc_name),
+                cmd_ctx.error_format)
+
+        # Set the remaining feature-specific parameters
+        properties['storage-volume-uri'] = storage_volume.uri
+        properties['timeout'] = timeout
```

Comment:
I got the following error when this line was not commented...
`Error: HTTPError: 400,14: Invalid value for query parameter timeout was specified.  [POST /api/partitions/.../operations/start-dump-program]`

Reply:
After reading through the HMC WS API documentation for Start Dump Program, it seems to me that the only way the timeout parameter can possibly be invalid is that the target volume was not a FICON volume, for which the timeout parameter is documented as invalid. Strange that it uses reason code 14 though, because that means error in the query parameters, and timeout is part of the body.

Anyway, I have changed the code to include the timeout parameter only when the target volume type (actually the type of its SG) is FICON. Please test again with the same volume type that caused the error.